pkg/daemon/ceph/client/subvolumegroup.go

```diff
@@ -172,34 +172,36 @@ func PinCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterInfo
 
 func validatePinningValues(pinning cephv1.CephFilesystemSubVolumeGroupSpecPinning) error {
 	numNils := 0
-	var err error
+
 	if pinning.Export != nil {
 		numNils++
 		if *pinning.Export > 256 {
```

Comment:
> Bit besides the point but out of curiosity I wonder is if this validation will ever fail in that sense these same values are already validate at the crd level so it should not let anyone apply a configuration that breaks the conditions.

@parth-gr No need for these conditions for invalid pinning values, right? Seems like the CRD validation catches all these errors so we don't need this validation in code. Was there a need for this?

Reply:
+1 These should be covered in CRD CEL validation which is already covered here https://github.com/rook/rook/blob/8aa89dd92c9a458978929b8024992abd682d3558/pkg/apis/ceph.rook.io/v1/types.go#L3032-L3045

Reply:
@parth-gr How about a separate PR to remove the unnecessary checks?

Reply:
In my opinion at least separate PR would be good as it is not exactly related to the current topic! I can go ahead and create the Issue and PR for this, if that is okay for you? Once @parth-gr has weighted in

Reply:
@travisn @Madhu-1 this is mainly because if someone  is using rook as an API, we had a discussion before with @BlaineEXE and we agreed to have these checks still as it is important.   https://github.com/rook/rook/pull/13283#discussion_r1419287617

Reply:
@parth-gr okay I can see so if someone would use the client package without the operator. I could add a comment to the function explaining this so it is clear for next reader of the code!